workflow/controller/workflowpod.go

```diff
@@ -615,6 +615,14 @@ func isResourcesSpecified(ctr *apiv1.Container) bool {
 
 // addMetadata applies metadata specified in the template
 func (woc *wfOperationCtx) addMetadata(pod *apiv1.Pod, tmpl *wfv1.Template, opts *createWorkflowPodOpts) {
+	// inherit workflow annotations and labels
+	for k, v := range woc.wf.ObjectMeta.Annotations {
```

Comment:
I'm not sure we just want to copy the workflows annotations and labels to the pods. Instead, perhaps a new spec.podMetadata field might be better - @jessesuen ?

Reply:
I agree. We probably don't want to inherit directly. Perhaps we could add something similar to podSpecPatch, e.g. podMetadataPatch.